sdk/communication/chat/src/main/java/com/azure/android/communication/chat/implementation/AzureCommunicationChatServiceImpl.java

```diff
@@ -0,0 +1,2309 @@
+            serviceClient.listChatReadReceiptsNext(pageId, callback);
+        }
+    }
+
+    private static final class ReadReceiptPageResponseRetriever extends PagedDataResponseRetriever<ReadReceipt, Page<ReadReceipt>> {
+        private final String chatThreadId;
+
+        private final AzureCommunicationChatServiceImpl serviceClient;
+
+        public ReadReceiptPageResponseRetriever(String chatThreadId, AzureCommunicationChatServiceImpl serviceClient) {
+            this.chatThreadId = chatThreadId;
+            this.serviceClient = serviceClient;
+        }
+
+        public Response<Page<ReadReceipt>> getFirstPage() {
+             return serviceClient.listChatReadReceiptsWithRestResponse(chatThreadId);
+        }
+
+        public Response<Page<ReadReceipt>> getPage(String pageId) {
+            return serviceClient.listChatReadReceiptsNextWithRestResponse(pageId);
+        }
+    }
+
+    private static final class ReadReceiptPageRetriever extends PagedDataRetriever<ReadReceipt, Page<ReadReceipt>> {
+        private final String chatThreadId;
+
+        private final AzureCommunicationChatServiceImpl serviceClient;
+
+        public ReadReceiptPageRetriever(String chatThreadId, AzureCommunicationChatServiceImpl serviceClient) {
+            this.chatThreadId = chatThreadId;
+            this.serviceClient = serviceClient;
+        }
+
+        public Page<ReadReceipt> getFirstPage() {
+             return serviceClient.listChatReadReceiptsWithRestResponse(chatThreadId).getValue();
+        }
+
+        public Page<ReadReceipt> getPage(String pageId) {
+            return serviceClient.listChatReadReceiptsNextWithRestResponse(pageId).getValue();
+        }
+    }
+
+    /**
+     * Gets read receipts for a thread.
+     * 
+     * @param chatThreadId Thread id to get the read receipts for.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return read receipts for a thread.
+     */
+    public Response<Page<ReadReceipt>> listChatReadReceiptsWithRestResponse(String chatThreadId) {
+        final retrofit2.Response<ResponseBody> response = this.executeRetrofitCall(service.listChatReadReceipts(chatThreadId, this.getApiVersion()));
+        if (response.isSuccessful()) {
+            if (response.code() == 200) {
+                final ReadReceiptsCollection decodedResult;
+                try {
+                    decodedResult = this.deserializeContent(response.headers(), response.body(), ReadReceiptsCollection.class);
+                } catch(Exception ex) {
+                    final String strContent = this.readAsString(response.body());
+                    throw new ErrorException(strContent, response.raw());
+                }
+                return new Response<>(response.raw().request(),
```

Comment:
nit; Indentation turned out a little weird in methods like this one..

Reply:
Agreed. Will improve in the codegen